scripts/generate_skeleton/generate_skeleton.py

```diff
@@ -0,0 +1,440 @@
+#!/usr/bin/env python3
+r"""Kubeflow Pipelines Component/Pipeline Skeleton Generator
+
+This script creates skeleton component or pipeline structures based on the
+CONTRIBUTING.md guide. It generates all required files with proper templates.
+
+Usage:
+    python scripts/generate_skeleton/generate_skeleton.py --type=component \\
+        --category=data_processing --name=my_processor
+    python scripts/generate_skeleton/generate_skeleton.py --type=pipeline \\
+        --category=ml_workflows --name=my_training_pipeline
+"""
+
+import argparse
+import keyword
+import os
+import sys
+from datetime import datetime, timezone
+from pathlib import Path
+
+import jinja2
+
+
+def _get_template_env() -> jinja2.Environment:
+    """Get Jinja2 environment with template loader."""
+    template_dir = Path(__file__).parent / "templates"
+    return jinja2.Environment(
+        loader=jinja2.FileSystemLoader(template_dir),
+        trim_blocks=True,
+        lstrip_blocks=True,
+    )
+
+
+def validate_name(name: str) -> None:
+    """Validate component/pipeline name for security and Python compatibility.
+
+    Args:
+        name: The name to validate
+
+    Raises:
+        ValueError: If name is invalid
+    """
+    if not name:
+        raise ValueError("Name cannot be empty")
+
+    # Check for directory traversal attempts
+    if "/" in name or "\\" in name:
+        raise ValueError("Name cannot contain path separators (/, \\)")
+
+    if "." in name:
+        raise ValueError("Name cannot contain dots (.)")
+
+    # Check for Python identifier validity
+    if not name.isidentifier():
+        raise ValueError(f"Name '{name}' is not a valid Python identifier")
+
+    # Check for Python keywords
+    if keyword.iskeyword(name):
+        raise ValueError(f"Name '{name}' is a Python keyword and cannot be used")
```

Comment:
`keyword.iskeyword()` only catches hard keywords. Consider also checking for soft keywords introduced in Python 3.10+ using `keyword.issoftkeyword()`:

```suggestion
    if keyword.iskeyword(name) or keyword.issoftkeyword(name):
        raise ValueError(f"Name '{name}' is a Python keyword and cannot be used")
```

Without this, users could create components named `match`, `case`, or `type` which are contextual keywords in modern Python.